packages/mongodb-memory-server-core/src/util/MongoBinaryDownload.ts

```diff
@@ -353,126 +353,205 @@ export class MongoBinaryDownload {
   }
 
   /**
-   * Downlaod given httpOptions to tempDownloadLocation, then move it to downloadLocation
+   * Download given httpOptions to tempDownloadLocation, then move it to downloadLocation
+   * @param url The URL to download the file from
    * @param httpOptions The httpOptions directly passed to https.get
    * @param downloadLocation The location the File should be after the download
    * @param tempDownloadLocation The location the File should be while downloading
+   * @param maxRetries Maximum number of retries on download failure
+   * @param baseDelay Base delay in milliseconds for retrying the download
    */
   async httpDownload(
     url: URL,
     httpOptions: RequestOptions,
     downloadLocation: string,
-    tempDownloadLocation: string
+    tempDownloadLocation: string,
+    maxRetries: number = 3,
+    baseDelay: number = 1000
   ): Promise<string> {
     log('httpDownload');
     const downloadUrl = this.assignDownloadingURL(url);
 
-    const maxRedirects = parseInt(resolveConfig(ResolveConfigVariables.MAX_REDIRECTS) || '');
+    const maxRedirects = parseInt(resolveConfig(ResolveConfigVariables.MAX_REDIRECTS) ?? '');
     const useHttpsOptions: Parameters<typeof https.get>[1] = {
       maxRedirects: Number.isNaN(maxRedirects) ? 2 : maxRedirects,
       ...httpOptions,
     };
 
+    for (let attempt = 0; attempt <= maxRetries; attempt++) {
+      try {
+        return await this.attemptDownload(
+          url,
+          useHttpsOptions,
+          downloadLocation,
+          tempDownloadLocation,
+          downloadUrl,
+          httpOptions
+        );
+      } catch (error) {
+        if (attempt === maxRetries) {
+          throw error;
+        }
+
+        const delay = baseDelay * Math.pow(2, attempt);
+        log(`httpDownload: attempt ${attempt + 1} failed, retrying in ${delay}ms...`);
+        await new Promise((resolve) => setTimeout(resolve, delay));
+      }
+    }
```

Comment:
This likely should not retry on cases like `404`.
Maybe add a new property `should_retry` to `DownloadError`?

Reply:
I added retryableStatusCodes, retryableErrorCodes to better control the retry logic.